### ISSUE TITLE
`no-name-in-module` not recognizing package with same base namespace

### ISSUE THREAD

**Issue by l0lawrence** [pylint-dev/pylint#8980]
### Bug description

within azure-monitor-opentelemetry package:

from azure.monitor.opentelemetry.exporter import (
    ApplicationInsightsSampler,
    AzureMonitorLogExporter,
    AzureMonitorMetricExporter,
    AzureMonitorTraceExporter,
)

This is raising a `no-name-in-module` warning even though `azure-monitor-opentelemetry-exporter` has also been imported. 

### Configuration

```ini
[testenv:pylint]
description=Lints a package with pylint (version {[testenv:pylint]pylint_version})
basepython = python37
pylint_version=2.15.8
```

### Command used

```shell
tox run pylint --root PACKAGE_LOCATION
```

### Pylint output

```shell
************* Module azure.monitor.opentelemetry._configure
azure\monitor\opentelemetry\_configure.py:37: [E0611(no-name-in-module), ] No name 'exporter' in module 'azure.monitor.opentelemetry'
```

### Expected behavior

I would expect pylint to recognize azure-monitor-opentelemetry-exporter as a package. 

### Pylint version

```shell
pylint 2.15.8
```

### OS / Environment

windows, python 3.7

**Comment by Pierre-Sassoulas** [comment]
Thank you for opening the issue. What is the version of the installed ``azure-monitor-opentelemetry``, 1.0.0b15 ?

**Comment by l0lawrence** [comment]
Hello! The issue was noticed when running pylint in the azure-sdk-for-python repo on the unreleased version [1.0.0b16](https://github.com/Azure/azure-sdk-for-python/tree/main/sdk/monitor/azure-monitor-opentelemetry)
